azure-pipelines/main-pipeline.yml

```diff
@@ -163,62 +184,61 @@ stages:
 
       condition: eq(variables['${{ service.name }}'], 'True')
       steps:
+      
+      # install JDK 16
+      
+      - script: ./azure-pipelines/java-16-install.sh
+        name: Install_JDK_16
+      - task: JavaToolInstaller@0
+        inputs:
+          versionSpec: '16'
+          jdkArchitectureOption: 'x64'
+          jdkSourceOption: 'LocalDirectory'
+          jdkFile: './jdk-16.0.2_linux-x64_bin.tar.gz'
+          jdkDestinationDirectory: './jdk'
+          cleanDestinationDirectory: true
```

Comment:
we need both of these steps?
(also, have consistent spacing between steps)
(applies to above at line ~120 also)

Reply:
Both of these steps are necessary. One is downloading the JDK. One is installing it with the tool. Clarified by naming, and spaced the tasks appropriately.